src/Core/Domain/IRepositories/IRefreshTokenRepository.cs

```diff
@@ -0,0 +1,16 @@
+﻿using CleanTemplate.Domain.Entities.Users;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CleanTemplate.Domain.IRepositories
+{
+    public interface IRefreshTokenRepository : IRepository<RefreshToken>
+    {
+        Task UpsertRefreshToken(RefreshToken refreshToken, CancellationToken cancellationToken);
+        Task<bool> ValidateRefreshToken(RefreshToken refreshToken, CancellationToken cancellationToken);
```

Comment:
1. Add Async at the end of name for async methods. For example: ValidateRefreshTokenAsync
2. For Upsert, my suggestion is using AddOrUpdate